phpunit/class-gutenberg-rest-post-counts-controller-test.php

```diff
@@ -0,0 +1,213 @@
+				'capability_type'    => 'post',
+			)
+		);
+	}
+
+	public function tear_down() {
+		unregister_post_type( 'private-cpt' );
+		parent::tear_down();
+	}
+
+	/**
+	 * @covers Gutenberg_REST_Post_Counts_Controller::register_routes
+	 */
+	public function test_register_routes() {
+		$routes = rest_get_server()->get_routes();
+		$this->assertArrayHasKey( '/wp/v2/counts/(?P<post_type>[\w-]+)', $routes );
+	}
+
+	public function test_context_param() {
+		// Single.
+		$request  = new WP_REST_Request( 'OPTIONS', '/wp/v2/counts/post' );
+		$response = rest_get_server()->dispatch( $request );
+		$data     = $response->get_data();
+		$this->assertSame( 'view', $data['endpoints'][0]['args']['context']['default'] );
+		$this->assertSame( array( 'view', 'edit' ), $data['endpoints'][0]['args']['context']['enum'] );
+	}
+
+	/**
+	 * @covers Gutenberg_REST_Post_Counts_Controller::et_item_schema
+	 */
+	public function test_get_item_schema() {
+		$request    = new WP_REST_Request( 'OPTIONS', '/wp/v2/counts/post' );
+		$response   = rest_get_server()->dispatch( $request );
+		$data       = $response->get_data();
+		$properties = $data['schema']['properties'];
+
+		$this->assertCount( 6, $properties );
+		$this->assertArrayHasKey( 'publish', $properties );
+		$this->assertArrayHasKey( 'future', $properties );
+		$this->assertArrayHasKey( 'draft', $properties );
+		$this->assertArrayHasKey( 'pending', $properties );
+		$this->assertArrayHasKey( 'private', $properties );
+		$this->assertArrayHasKey( 'trash', $properties );
+	}
+
+	/**
+	 * @covers Gutenberg_REST_Post_Counts_Controller::get_item
+	 */
+	public function test_get_item_response() {
+		wp_set_current_user( self::$admin_id );
+		$request  = new WP_REST_Request( 'GET', '/wp/v2/counts/post' );
+		$response = rest_get_server()->dispatch( $request );
+		$data     = $response->get_data();
+
+		$this->assertSame( 200, $response->get_status() );
+		$this->assertArrayHasKey( 'publish', $data );
+		$this->assertArrayHasKey( 'future', $data );
+		$this->assertArrayHasKey( 'draft', $data );
+		$this->assertArrayHasKey( 'pending', $data );
+		$this->assertArrayHasKey( 'private', $data );
+		$this->assertArrayHasKey( 'trash', $data );
+	}
+
+	/**
+	 * @covers Gutenberg_REST_Post_Counts_Controller::get_item
+	 */
+	public function test_get_item() {
+		wp_set_current_user( self::$admin_id );
+
+		$published = self::factory()->post->create( array( 'post_status' => 'publish' ) );
+		$future    = self::factory()->post->create(
+			array(
+				'post_status' => 'future',
+				'post_date'   => gmdate( 'Y-m-d H:i:s', strtotime( '+1 day' ) ),
+			)
+		);
+		$draft     = self::factory()->post->create( array( 'post_status' => 'draft' ) );
+		$pending   = self::factory()->post->create( array( 'post_status' => 'pending' ) );
+		$private   = self::factory()->post->create( array( 'post_status' => 'private' ) );
+		$trashed   = self::factory()->post->create( array( 'post_status' => 'trash' ) );
```

Comment:
Note to self:

I couldn't work out how to unit test custom post statuses registered by `register_post_status`. 

I suspect because the post statuses are stored in `$GLOBALS` in Core, ([example](https://github.com/WordPress/wordpress-develop/blob/6.6.2/src/wp-includes/post.php#L1516)) and that object is different in the test environment. 

Not sure. Works in Core unit tests.